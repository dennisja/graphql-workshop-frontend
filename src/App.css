.App {
    text-align: center;
}

.App-logo {
    animation: App-logo-spin infinite 20s linear;
    height: 80px;
}

.App-header {
    background-color: green;
    height: 150px;
    padding: 20px;
    color: white;
}

.App-navigation a, .App-navigation a:visited {
    text-decoration: none;
    color: white;
    display: inline-block;
    padding: 5px;
}

.active, active a{
  color: green !important;
  background-color: white
}

.App-title {
    font-size: 1.5em;
}

.App-intro {
    font-size: large;
}

@keyframes App-logo-spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}