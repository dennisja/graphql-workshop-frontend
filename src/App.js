import React, { Component } from "react";
import { ApolloProvider } from "react-apollo";
import { BrowserRouter } from "react-router-dom";
import apolloClient from "./components/ApolloClient";
import logo from "./logo.svg";
import "./App.css";
import Routes from './components/Routes';

class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <ApolloProvider client={apolloClient}>
          <div className="App">
            <header className="App-header">
              <img src={logo} className="App-logo" alt="logo" />
              <h1 className="App-title">Welcome to React</h1>
            </header>
            <div>
              <Routes />
            </div>
          </div>
        </ApolloProvider>
      </BrowserRouter>
    );
  }
}

export default App;
