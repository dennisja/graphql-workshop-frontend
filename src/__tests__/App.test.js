import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { MockedProvider } from "react-apollo/test-utils";
import { StaticRouter, MemoryRouter } from "react-router-dom";

import App from "../App";

describe("App component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <StaticRouter context={{}}>
        <MockedProvider>
          <App />
        </MockedProvider>
      </StaticRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders properly", () => {
    const wrapper = shallow(
      <MockedProvider>
        <App />
      </MockedProvider>
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
