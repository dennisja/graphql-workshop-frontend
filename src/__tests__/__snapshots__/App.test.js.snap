// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders properly 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "__operations_cache__": Map {
        "getLinks" => Object {
          "query": Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "getLinks",
                },
                "operation": "query",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "links",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "url",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "title",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 73,
              "start": 0,
            },
          },
          "variables": undefined,
        },
      },
      "cache": InMemoryCache {
        "addTypename": true,
        "config": Object {
          "addTypename": true,
          "dataIdFromObject": [Function],
          "fragmentMatcher": HeuristicFragmentMatcher {},
          "storeFactory": [Function],
        },
        "data": ObjectCache {
          "data": Object {},
        },
        "optimistic": Array [],
        "silenceBroadcast": false,
        "watches": Array [],
      },
      "defaultOptions": Object {},
      "disableNetworkFetches": false,
      "link": ApolloLink {
        "request": [Function],
      },
      "mutate": [Function],
      "query": [Function],
      "queryDeduplication": true,
      "queryManager": QueryManager {
        "dataStore": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "storeFactory": [Function],
            },
            "data": ObjectCache {
              "data": Object {},
            },
            "optimistic": Array [],
            "silenceBroadcast": false,
            "watches": Array [],
          },
        },
        "deduplicator": ApolloLink {
          "request": [Function],
        },
        "fetchQueryPromises": Map {
          "2" => Object {
            "promise": undefined,
            "reject": [Function],
            "resolve": [Function],
          },
        },
        "idCounter": 3,
        "link": ApolloLink {
          "request": [Function],
        },
        "mutationStore": MutationStore {
          "store": Object {},
        },
        "onBroadcast": [Function],
        "queries": Map {},
        "queryDeduplication": true,
        "queryIdsByName": Object {
          "getLinks": Array [],
        },
        "queryStore": QueryStore {
          "store": Object {},
        },
        "scheduler": QueryScheduler {
          "inFlightQueries": Object {},
          "intervalQueries": Object {},
          "pollingTimers": Object {},
          "queryManager": [Circular],
          "registeredQueries": Object {},
          "ssrMode": false,
        },
      },
      "reFetchObservableQueries": [Function],
      "resetStore": [Function],
      "resetStoreCallbacks": Array [],
      "ssrMode": false,
      "store": DataStore {
        "cache": InMemoryCache {
          "addTypename": true,
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "storeFactory": [Function],
          },
          "data": ObjectCache {
            "data": Object {},
          },
          "optimistic": Array [],
          "silenceBroadcast": false,
          "watches": Array [],
        },
      },
      "version": "2.2.8",
      "watchQuery": [Function],
    }
  }
>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <h1
        className="App-title"
      >
        Welcome to React
      </h1>
    </header>
    <div>
      <Apollo(Component) />
    </div>
  </div>
</ApolloProvider>
`;
